crates/cubecl-core/src/runtime_tests/plane.rs

```diff
@@ -75,132 +74,296 @@ pub fn kernel_broadcast<F: Float>(output: &mut Tensor<F>) {
+    input[16] = 999.0; // I don't want the max to always be the last element.
+
+    let mut expected = input.clone();
+
+    for k in 1..plane_size as usize {
+        for v in 0..vectorization as usize {
+            expected[v] = expected[v].max(input[v + k * vectorization as usize]);
+        }
+    }
+    let input: Vec<F> = input.into_iter().map(|x| F::new(x)).collect();
+    let expected: Vec<F> = expected.into_iter().map(|x| F::new(x)).collect();
+
     test_plane_operation::<TestRuntime, F, _>(
-        as_type![F: 4.0, 5.0, 7.0, 1.0],
-        as_type![F: 7.0, 5.0, 7.0, 1.0],
+        &input,
+        &expected,
+        vectorization,
         client.clone(),
-        |cube_dim, settings, handle| {
-            kernel_max::launch::<F, TestRuntime>(&client, cube_dim, settings, handle)
+        |cube_count, handle| {
+            kernel_max::launch::<F, TestRuntime>(
+                &client,
+                cube_count,
+                CubeDim::new(32, 1, 1),
+                handle,
+            )
         },
     );
 }
 
 pub fn test_plane_min<TestRuntime: Runtime, F: Float + CubeElement>(
     client: ComputeClient<TestRuntime::Server, TestRuntime::Channel>,
+    vectorization: u8,
 ) {
+    let plane_size = 32;
+    let mut input: Vec<f32> = (0..plane_size * vectorization as u32)
+        .map(|x| x as f32)
+        .collect();
+    input[16] = -5.0; // I don't want the min to always be the first element.
+
+    let mut expected = input.clone();
+
+    for k in 1..plane_size as usize {
+        for v in 0..vectorization as usize {
+            expected[v] = expected[v].min(input[v + k * vectorization as usize]);
+        }
+    }
+    let input: Vec<F> = input.into_iter().map(|x| F::new(x)).collect();
+    let expected: Vec<F> = expected.into_iter().map(|x| F::new(x)).collect();
+
     test_plane_operation::<TestRuntime, F, _>(
-        as_type![F: 4.0, 5.0, 7.0, 1.0],
-        as_type![F: 1.0, 5.0, 7.0, 1.0],
+        &input,
+        &expected,
+        vectorization,
         client.clone(),
-        |cube_dim, settings, handle| {
-            kernel_min::launch::<F, TestRuntime>(&client, cube_dim, settings, handle)
+        |cube_count, handle| {
+            kernel_min::launch::<F, TestRuntime>(
+                &client,
+                cube_count,
+                CubeDim::new(32, 1, 1),
+                handle,
+            )
         },
     );
 }
 
 pub fn test_plane_all<TestRuntime: Runtime, F: Float + CubeElement>(
     client: ComputeClient<TestRuntime::Server, TestRuntime::Channel>,
+    vectorization: u8,
 ) {
+    let plane_size = 32;
+    let mut input: Vec<f32> = (0..plane_size * vectorization as u32)
+        .map(|x| (x % 5) as f32) // the predicate is x < 5 which is always satisfied at this step.
+        .collect();
+
+    for k in 0..vectorization as usize {
+        if k % 2 == 0 {
+            input[4 * vectorization as usize + k] = 10.0; // Make all even batches false by setting an element to be > 5.
```

Comment:
What's the 4 for?

Reply:
I want to problematic entries to be in the middle. So, I push them by 4 * vectorization. I don't have a good justification, it just feel like a more robust test to me, but it is somewhat subjective.